### PR TITLE
Retain Rust code items from `mod` decorated with `subxt` attribute

### LINKED ISSUE #346: Retain Rust code items from `mod` decorated with `subxt` attribute
At the moment any Rust code items inside the decorated `mod` will be erased. 

e.g.

```
#[subxt::subxt(runtime_metadata_path = "metadata.scale")]
pub mod node_runtime { 
  pub struct MyStruct {} // this will be erased
}
```
See https://github.com/paritytech/subxt/commit/6306b9fc073b59133566cfabf9502c99d2703161

From #309

### PR DESCRIPTION
**Pull request by Xanewok**
Closes #346

### CODE REVIEW

File: testing/ui-tests/src/lib.rs [hunk 1]
```diff
@@ -50,4 +50,5 @@ fn ui_tests() {
 fn ui_fail() {
     let t = trybuild::TestCases::new();
     t.compile_fail("src/incorrect/*.rs");
+    t.pass("src/correct/*.rs");
```
Comment (lexnv) [inline testing/ui-tests/src/lib.rs:53]: Nit: Should we move this to the `ui_tests` category? Having it under `ui_fail` might be misleading

**Review by lexnv — APPROVED** [review]
👍

File: codegen/src/api/mod.rs [hunk 2]
```diff
@@ -324,9 +322,14 @@ impl RuntimeGenerator {
             })
             .collect();
 
+        let rust_items: Vec<_> = item_mod_ir.rust_items().collect();
```
Comment (ascjones) [inline codegen/src/api/mod.rs:325]:
```suggestion
        let rust_items = item_mod_ir.rust_items();
```
I think `quote!` can handle the iterator directly?

**Review by ascjones — APPROVED** [review]
LGTM

File: testing/ui-tests/src/lib.rs (hunk 1 above)
Comment (niklasad1) [reply testing/ui-tests/src/lib.rs:53]: or just move this one to `ui_tests` and keep `ui_fall`

**Review by niklasad1 — APPROVED** [review]
:+1:

File: testing/ui-tests/src/lib.rs (hunk 1 above)
Comment (Xanewok) [reply testing/ui-tests/src/lib.rs:53]: Ah, didn't spot the passing cases above; will move it

File: codegen/src/api/mod.rs (hunk 2 above)
Comment (Xanewok) [reply codegen/src/api/mod.rs:325]: Oh, it does! Good catch, thanks

**Pull request merged**